src/Sarif/Writers/SarifLogger.cs

```diff
@@ -223,6 +223,9 @@ public void Dispose()
                 _issueLogJsonWriter.Dispose();
             }
             if (_textWriter != null) { _textWriter.Dispose(); }
+
+            GC.SuppressFinalize(_issueLogJsonWriter);
+            GC.SuppressFinalize(_textWriter);
```

Comment:
This isn't the correct fix. I'm curious, does the warning go away when you add this code?

To understand why this is wrong, look at the web page for CA1816: [https://msdn.microsoft.com/en-us/library/ms182269.aspx](https://msdn.microsoft.com/en-us/library/ms182269.aspx). The idea is that the Dispose method for an object `X` frees the resources held be `X`. That means it's no longer necessary for the garbage collector to free `X`'s resources by calling `X`'s "finalizer" method. So you _tell_ the garbage collector not to bother "finalizing" `X` by calling `GC.SuppressFinalize` _on X itself_. So the correct fix is:

```
GC.SuppressFinalize(this);
```

You don't have to call `GC.SuppressFinalize` on `X`'s resources (in this case, `_textWriter` and `_issueLogJsonWriter`) because you just called _their_ `Dispose` methods -- and they are responsible for freeing their own resources.